dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/DockerfileGitHubUtil.java

```diff
@@ -358,7 +359,16 @@ public GitHubJsonStore getGitHubJsonStore(String store) {
 
     public void createPullReq(GHRepository origRepo,
                               String branch, GHRepository forkRepo,
-                              PullRequestInfo pullRequestInfo) throws InterruptedException, IOException {
+                              PullRequestInfo pullRequestInfo,
+                              RateLimiter rateLimiter) throws InterruptedException, IOException {
+
+        while (!rateLimiter.checkLimit()) {
+            //we have exceeded the limit and have no more tokens in the bucket
+            //wait for 10s before next try
+            log.info("We have exceeded the limit of PR creation per hour.. sleeping for 10s");
+            Thread.sleep(10000);
```

Comment:
We don't need this.  The rate limiter calls are blocking if a token is not available.  We should try to acquire a token and block until one is available instead of checking and then sleeping for 10 seconds just to sleep again.  See https://bucket4j.com/8.1.0/toc.html#using-bucket-as-throttler and https://bucket4j.com/8.1.0/toc.html#limiting-the-rate-of-access-to-rest-api as examples.